test/micro/org/openjdk/bench/java/io/DataOutputStreamTest.java

```diff
@@ -0,0 +1,141 @@
+        @Param({"4096"}) int SIZE;
+        @Param({"char", "short", "int", /*"string"*/}) String BASIC_TYPE;
+        final ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream(SIZE);
+        final File f = new File("DataOutputStreamTest.tmp");
+        String outputString;
+        FileOutputStream fileOutputStream;
+        DataOutput bufferedFileStream, rawFileStream, byteArrayStream;
+        BasicType basicType;
+
+        @Setup(Level.Trial)
+        public void setup() {
+            try {
+                fileOutputStream = new FileOutputStream(f);
+                byteArrayStream = new DataOutputStream(byteArrayOutputStream);
+                rawFileStream = new DataOutputStream(fileOutputStream);
+                bufferedFileStream = new DataOutputStream(new BufferedOutputStream(fileOutputStream));
+                switch (BASIC_TYPE.toLowerCase()) {
+                    case "char": basicType = BasicType.CHAR; break;
+                    case "short": basicType =  BasicType.SHORT; break;
+                    case "int": basicType = BasicType.INT; break;
+                    case "string": basicType = BasicType.STRING; break;
+                    default: throw new RuntimeException("Unhandled basic type:" + BASIC_TYPE);
+                };
+                outputString = new String(new byte[SIZE]);
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    public void writeChars(BenchmarkState state, DataOutput dataOutput) {
+        try {
+            for (int i = 0; i < state.SIZE; i += 2) {
+                dataOutput.writeChar(i);
+            }
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void writeShorts(BenchmarkState state, DataOutput dataOutput) {
+        try {
+            for (int i = 0; i < state.SIZE; i += 2) {
+                dataOutput.writeShort(i);
+            }
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void writeInts(BenchmarkState state, DataOutput dataOutput) {
+        try {
+            for (int i = 0; i < state.SIZE; i += 4) {
+                dataOutput.writeInt(i);
+            }
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void writeString(BenchmarkState state, DataOutput dataOutput) {
+        try {
+            dataOutput.writeChars(state.outputString);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void write(BenchmarkState state, DataOutput dataOutput) {
+        switch (state.basicType) {
+            case CHAR: writeChars(state, dataOutput); break;
+            case SHORT: writeShorts(state, dataOutput); break;
+            case INT: writeInts(state, dataOutput); break;
+            case STRING: writeString(state, dataOutput); break;
+        }
+    }
```

Comment:
So, the way I see it, this can be replaced by:

```
    public void write(DataOutput dataOutput) {
        switch (basicType) {
            case CHAR: 
                for (int i = 0; i < SIZE; i += 2) {
                    dataOutput.writeChar(i);
                }
                break;
            case SHORT: 
                for (int i = 0; i < SIZE; i += 2) {
                    dataOutput.writeShort(i);
                }
                break;
            case INT: 
                for (int i = 0; i < SIZE; i += 4) {
                    dataOutput.writeInt(i);
                }
                break;
            case STRING:
                dataOutput.writeChars(state.outputString);
                break;
        }
    }
```

...thus keeping all the code in one place, and simplifying the benchmark.

Reply:
I didn't quite do that, perferring to keep the writeX functions separate.

Reply:
No problem, this is a question of personal style. If this was a very fine-grained benchmark, we would worry about inlining/dispatch costs.